File: Server/src/main/java/lk/wnl/wijeya/entity/MaterialPorderMaterial.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "material_porder_material")
public class MaterialPorderMaterial {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "material_porder_id", nullable = false)
    private MaterialPorder materialPorder;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "material_id", nullable = false)
    private Material material;

    @Column(name = "quentity")
    private Integer quentity;

    @Column(name = "expected_line_cost", precision = 7, scale = 2)
    private BigDecimal expectedLineCost;

}

File: Client/src/app/entity/MaterialPorderMaterial.ts
import {Material} from "./Material";
import {MaterialPorder} from "./MaterialPorder";

export class MaterialPorderMaterial {


  id!: number;
  materialPorder!: MaterialPorder;
  material!: Material;
  quantity!: number;
  expectedLineCost!: number;

  constructor(id: number, material: Material, quentity: number, expectedLineCost: number) {
    this.id = id;
    this.material = material;
    this.quantity = quentity;
    this.expectedLineCost = expectedLineCost;
  }
}
